### PR TITLE
fix some bugs

### PR DESCRIPTION
**Pull request by stishkin**
- fix queries in timer retention
- do not discard proxy record after proxy state is processed, since that record needs to persist

### CODE REVIEW

File: src/ApiService/ApiService/Functions/TimerProxy.cs [hunk 1]
```diff
@@ -40,11 +42,14 @@ public class TimerProxy {
 
             if (VmStateHelper.NeedsWork.Contains(proxy.State)) {
                 _logger.Info($"scaleset-proxy: update state. proxy:{proxy.Region} state:{proxy.State}");
-                await proxyOperations.ProcessStateUpdate(proxy);
+                proxy = await proxyOperations.ProcessStateUpdate(proxy);
             }
 
-            if (proxy.State != VmState.Stopped && proxyOperations.IsOutdated(proxy)) {
-                await proxyOperations.Replace(proxy with { Outdated = true });
```
Comment (nharper285) [inline src/ApiService/ApiService/Functions/TimerProxy.cs:47]: This was the cause of the bad record replacements?

**Review by nharper285 — APPROVED** [review]

File: src/ApiService/ApiService/Functions/TimerProxy.cs (hunk 1 above)
Comment (stishkin) [reply src/ApiService/ApiService/Functions/TimerProxy.cs:47]: we are using a potentially outdated record here, so we could overwrite an updated one with older version.

File: src/ApiService/ApiService/onefuzzlib/VmOperations.cs [hunk 2]
```diff
@@ -218,7 +218,9 @@ await _context.Creds.GetResourceGroupResource()
                 extensionName,
                 extension
             );
-        } catch (RequestFailedException ex) when (ex.Status == 409 && ex.Message.Contains("VM is marked for deletion")) {
+        } catch (RequestFailedException ex) when
+            (ex.Status == 409 &&
+                (ex.Message.Contains("VM is marked for deletion") || ex.Message.Contains("The request failed due to conflict with a concurrent request."))) {
             _logTracer.Info(ex.Message);
```
Comment (tevoinea) [inline src/ApiService/ApiService/onefuzzlib/VmOperations.cs:224]:
```suggestion
            _logTracer.Info($"Tried to create extension but failed due to {ex.Message}");
```

So we know what the message is about

File: src/ApiService/ApiService/onefuzzlib/orm/Queries.cs [hunk 3]
```diff
@@ -46,12 +46,25 @@ public static string And(IEnumerable<string> queries)
...
+        }
+
         public static string TimeRange(DateTimeOffset min, DateTimeOffset max) {
             // NB: this uses the auto-populated Timestamp property, and will result in a table scan
             // TODO: should this be inclusive at the endpoints?
             return TableClient.CreateQueryFilter($"Timestamp lt {max} and Timestamp gt {min}");
         }
 
+        public static string NewerThan(DateTimeOffset t) {
+            return TableClient.CreateQueryFilter($"Timestamp gt {t}");
```
Comment (tevoinea) [inline src/ApiService/ApiService/onefuzzlib/orm/Queries.cs:61]:
```suggestion
            return TableClient.CreateQueryFilter($"Timestamp gt {t.ToString("o")}");
```

To keep the same behavior from `TimerRetention`

File: src/ApiService/ApiService/onefuzzlib/orm/Queries.cs [hunk 4]
```diff
@@ -46,12 +46,25 @@ public static string And(IEnumerable<string> queries)
...
+        }
+
+        public static string OlderThan(DateTimeOffset t) {
+            return TableClient.CreateQueryFilter($"Timestamp lt {t}");
```
Comment (tevoinea) [inline src/ApiService/ApiService/onefuzzlib/orm/Queries.cs:65]:
```suggestion
            return TableClient.CreateQueryFilter($"Timestamp lt {t.ToString("o")}");
```

Same here.

File: src/ApiService/ApiService/Functions/TimerProxy.cs [hunk 5]
```diff
@@ -40,11 +42,14 @@ public class TimerProxy {
...
+            if (proxy is not null && (proxy.State != VmState.Stopped && proxyOperations.IsOutdated(proxy))) {
```
Comment (tevoinea) [inline src/ApiService/ApiService/Functions/TimerProxy.cs:48]:
```suggestion
            if (proxy is not null && proxy.State != VmState.Stopped && proxyOperations.IsOutdated(proxy)) {
```

**Review by tevoinea — APPROVED** [review]

File: src/ApiService/ApiService/onefuzzlib/orm/Queries.cs (hunk 3 above)
Comment (stishkin) [reply src/ApiService/ApiService/onefuzzlib/orm/Queries.cs:61]: TableClient.CreateQueryFilter <- does the conversion based on type. So it's supposed to do t.ToString("o") implicitly

File: src/ApiService/ApiService/onefuzzlib/orm/Queries.cs (hunk 4 above)
Comment (stishkin) [reply src/ApiService/ApiService/onefuzzlib/orm/Queries.cs:65]: same as above

File: src/ApiService/ApiService/Functions/TimerProxy.cs (hunk 5 above)
Comment (stishkin) [reply src/ApiService/ApiService/Functions/TimerProxy.cs:48]: ok

**Pull request merged**